src/drc/api/serializers.py

```diff
@@ -58,7 +59,15 @@ def to_representation(self, file):
         url_field = self.parent.fields["url"]
         lookup_field = url_field.lookup_field
         kwargs = {lookup_field: getattr(model_instance, lookup_field)}
-        return reverse(self.view_name, kwargs=kwargs, request=request)
+        url = reverse(self.view_name, kwargs=kwargs, request=request)
+
+        # Retrieve the correct version to construct the download url that
+        # points to the content of that version
+        if hasattr(self.parent.instance, 'versie'):
+            versie = self.parent.instance.versie
+        else:
+            versie = self.parent.instance.get(uuid=kwargs['uuid']).versie
+        return f'{url}/download?versie={versie}'
```

Comment:
what's with the hardcoding urls? Always reverse URLs.

It's also a bad practice to construct querystring arguments manually, you should use `django.utils.http.urlencode` and construct the URL like that.

Reply:
Additional thought: does the `versie` querystring parameter show up in the API spec? Because it definitely should, so we'll need to add that to the schema for the `download` viewset action